File: 1297-maximum-number-of-balloons/maximum-number-of-balloons.py
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        balloon = Counter("balloon")
        word = Counter(text)

        res = float("inf")
        for c in balloon:
            if c in word:
                res = min(res, word[c] // balloon[c])
            else:
                return 0
        return res
        

File: 1297-maximum-number-of-balloons/maximum-number-of-balloons.cpp
class Solution {
public:
    int maxNumberOfBalloons(string text) {
        int res = INT_MAX;
        unordered_set<char> balloon = {'b', 'a', 'l', 'o', 'n'};
        unordered_map<char, int> mp;
        for (char c : text) {
            if (balloon.contains(c)) {
                mp[c]++;
            }
        }

        if (mp.size() < 5) return 0;

        mp['l'] /= 2;
        mp['o'] /= 2;

        for (const auto& [c, freq] : mp) {
            res = min(res, freq);
        }

        return res;
        
    }
};